server/src/UseCases/Games/Create/CreateGameCommandHandler.cs

```diff
@@ -1,29 +1,31 @@
 ﻿using System.Threading;
 using System.Threading.Tasks;
 using BGS.ApplicationCore.Entities;
+using BGS.ApplicationCore.Games.Interfaces;
 using BGS.SharedKernel;
 using BGS.SharedKernel.Results;
 using MediatR;
 
 namespace BGS.UseCases.Games.Create;
 
-public class CreateGameCommandHandler : IRequestHandler<CreateGameCommand, Result>
+public class CreateGameCommandHandler(
+        IRepository<Game> gameRepository,
+        IGameDuplicationChecker gameDuplicationChecker)
+    : IRequestHandler<CreateGameCommand, Result>
 {
-    private readonly IRepository<Game> _gameRepository;
-
-    public CreateGameCommandHandler(IRepository<Game> gameRepository)
-    {
-        _gameRepository = gameRepository;
-    }
-
     public async Task<Result> Handle(CreateGameCommand command, CancellationToken cancellationToken)
     {
+        if (await gameDuplicationChecker.Check(command.GameName))
+        {
+            return Result<Result>.Error($"Game with the name {command.GameName} is already exists.");
```

Comment:
this line is duplicated here and in `UpdateGameCommandHandler`. Let's change return type of `IGameDuplicationChecker` to Result, and don't use `Result<Result>` just create error method in Result.